test/e2e/wait_timeout_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestWaitTimeout(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+
+	yaml1 := `
+ apiVersion: batch/v1 
+ kind: Job 
+ metadata: 
+   name: successful-job
+ spec: 
+   template: 
+     metadata: 
+       name: successful-job 
+     spec: 
+       containers: 
+       - name: successful-job 
+         image: busybox 
+         command: ["/bin/sh", "-c", "sleep 5"] 
+       restartPolicy: Never
+`
+
+	name := "test-wait-timeout"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+	}
+
+	cleanUp()
+	defer cleanUp()
+
+	logger.Section("Resource timed out waiting", func() {
+		_, err := kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name, "--wait-timeout",
+			"2m", "--wait-resource-timeout", "3s", "--json"},
```

Comment:
how come we are using different timeout for different purposes? lets standardize...

1s - for failing timeout
100s - for cases when resource succeeds before timeout
10000s - for cases when specific timeout should not play a role

Reply:
Sure done